src/util/system.cpp

```diff
@@ -198,30 +190,98 @@ static util::SettingsValue InterpretOption(std::string& section, std::string& ke
     }
     if (key.substr(0, 2) == "no") {
         key.erase(0, 2);
-        // Double negatives like -nofoo=0 are supported (but discouraged)
-        if (!InterpretBool(value)) {
-            LogPrintf("Warning: parsed potentially confusing double-negative -%s=%s\n", key, value);
-            return true;
-        }
         return false;
     }
-    return value;
+    return true;
 }
 
 /**
- * Check settings value validity according to flags.
+ * Interpret settings value and apply ALLOW_* type checking flags.
  *
- * TODO: Add more meaningful error checks here in the future
- * See "here's how the flags are meant to behave" in
- * https://github.com/bitcoin/bitcoin/pull/16097#issuecomment-514627823
+ * Return an error string and nullopt if an invalid value was provided that
+ * isn't allowed by the flags, otherwise return the parsed value.
  */
-static bool CheckValid(const std::string& key, const util::SettingsValue& val, unsigned int flags, std::string& error)
+static Optional<util::SettingsValue> InterpretValue(const std::string& key,
+    const std::string* value,
```

Comment:
nit, I've changed this to `Optional<string>`, see https://github.com/promag/bitcoin/commit/71da58bc9437b977e56029bd8d2516bcf61cb823.

Reply:
Yeah don't bother